Tests/MatterControl.Tests/MatterControl/MeshCsgTests.cs

```diff
@@ -0,0 +1,240 @@
+   and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+The views and conclusions contained in the software and documentation are those
+of the authors and should not be interpreted as representing official policies,
+either expressed or implied, of the FreeBSD Project.
+*/
+#define DEBUG_INTO_TGAS
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using MatterHackers.Agg;
+using MatterHackers.Agg.Platform;
+using MatterHackers.DataConverters3D;
+using MatterHackers.MatterControl.DesignTools;
+using MatterHackers.MatterControl.DesignTools.Operations;
+using MatterHackers.MatterControl.Tests.Automation;
+using MatterHackers.PolygonMesh.Csg;
+using MatterHackers.PolygonMesh.Processors;
+using MatterHackers.VectorMath;
+using NUnit.Framework;
+
+namespace MatterHackers.PolygonMesh.UnitTests
+{
+	[TestFixture, Category("Agg.PolygonMesh.Csg")]
+	public class MeshCsgTests
+	{
+		//[Test]
+		public void CylinderMinusCylinder()
+		{
+			AggContext.StaticData = new FileSystemStaticData(TestContext.CurrentContext.ResolveProjectPath(4, "StaticData"));
+			MatterControlUtilities.OverrideAppDataLocation(TestContext.CurrentContext.ResolveProjectPath(4));
+
+			AggContext.StaticData = new FileSystemStaticData(TestContext.CurrentContext.ResolveProjectPath(5, "MatterControl", "StaticData"));
+			MatterControlUtilities.OverrideAppDataLocation(TestContext.CurrentContext.ResolveProjectPath(5));
+
+			int sides = 3;
+			IObject3D keep = CylinderAdvancedObject3D.Create(20, 20, sides);
+			IObject3D subtract = CylinderAdvancedObject3D.Create(10, 21, sides);
+			IObject3D subtractCentered = new SetCenter(subtract, keep.GetCenter());
+
+			var keepMesh = keep.Mesh;
+			var subtractMesh = subtract.Mesh;
+			subtractMesh.Transform(subtract.WorldMatrix());
+
+			var split1 = new DebugFace()
+			{
+				EvaluateHeight = 10,
+				FileName = "Split1"
+			};
+
+			var resultMesh = keepMesh.Subtract(subtractMesh, null, CancellationToken.None,
+				split1.Split, split1.Result);
+
+			split1.FinishOutput();
+
+			resultMesh.Save("c:/temp/mesh.stl", CancellationToken.None);
```

Comment:
This pathing style isn't promoted or used on Windows. It seemingly works because I keep seeing it checked in but it breaks with convention and the lineage of the backslash path separator on Windows